incubator/elasticsearch/templates/deployment.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: "{{ template "fullname" . }}"
+  labels:
+    app: "{{ template "name" . }}"
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  template:
+    metadata:
+      labels:
+        app: "{{ template "name" . }}"
+        release: "{{ .Release.Name }}"
+      annotations:
+        pod.beta.kubernetes.io/init-containers: '[
```

Comment:
the past two version (1.6, 1.7) support the `initContainers` key in the PodSpec, so I think we can use that, unless you think we need to support 1.5 and less?